pkg/plugins/git/generic/main.go

```diff
@@ -71,51 +94,54 @@ func Checkout(branch, remoteBranch, workingDir string) error {
 		})
 
 		if err != nil {
+			logrus.Debugf("Branch: '%v' - \n\t%v", branch, err)
 			return err
 		}
 
-		logrus.Debugf("Checkout branch: '%v'", remoteBranch)
 		// Checkout locale branch without creating it yet
 		err = w.Checkout(&git.CheckoutOptions{
 			Branch: plumbing.NewBranchReferenceName(remoteBranch),
-			Create: false,
+			Create: true,
 			Keep:   false,
 			Force:  true,
 		})
 
-		// Branch doesn't exist locally
-		if err == plumbing.ErrReferenceNotFound {
-			logrus.Debugf("Creating first branch: '%v'", remoteBranch)
-
-			err = w.Checkout(&git.CheckoutOptions{
-				Branch: plumbing.NewBranchReferenceName(remoteBranch),
-				Create: true,
-				Keep:   false,
-				Force:  true,
-			})
+		logrus.Debugf("Creating branch '%v'", remoteBranch)
 
-			if err != nil &&
-				(err != plumbing.ErrReferenceNotFound ||
-					err != git.ErrBranchExists) {
-				return err
-			}
-
-		} else if err != nil && err != plumbing.ErrReferenceNotFound {
+		if err != nil {
 			return err
 		}
+
 	} else if err != plumbing.ErrReferenceNotFound && err != nil {
+		logrus.Debugln(err)
 		return err
 	} else {
+
 		// Means that a local branch named remoteBranch already exist
 		// so we want to be sure that the local branch is
 		// aligned with the remote one.
-		remoteBranchRef := fmt.Sprintf("refs/remotes/origin/%s", remoteBranch)
 
-		logrus.Infof(remoteBranchRef)
+		remote, err := r.Remote("origin")
+		if err != nil {
+			return err
+		}
+		refs, err := remote.List(&git.ListOptions{})
+
+		if !exists(
+			plumbing.NewBranchReferenceName(remoteBranch),
+			refs) {
+			logrus.Debugf("No remote name '%v'", remoteBranch)
+			return nil
+		}
+
+		remoteBranchRef := plumbing.NewRemoteReferenceName("origin", remoteBranch)
+
+		logrus.Infof("Remote branch: ", remoteBranchRef)
 
 		remoteRef, err := r.Reference(
 			plumbing.ReferenceName(
 				remoteBranchRef), true)
```

Comment:
```suggestion
		remoteRef, err := r.Reference(remoteBranchRef, true)
```

Honestly I have no idea why the linter complains about this instruction here, but not on the main branch o_0

```
pkg/plugins/git/generic/main.go:142:26: unnecessary conversion (unconvert)
                        plumbing.ReferenceName(
                                              ^
```

Reply:
Probably because remoteBranchRef is already a reference, I'll fix that

Reply:
I could directly use remoteRef,err := r.Reference(remoteBranchRef, true) as remoteBranchRef as already a ReferenceName